### PR TITLE
Fixed PanOS empty config parsing

### PR DESCRIPTION
**Pull request by joewesch**
Closes: #803 

The `PaloAltoNetworksConfigParser` was written contradictory to the rest of the parsers in that it raised an error when the configuration was empty. This PR changes that behavior to bring it in line with the rest of the parsers. I also added a unit test to verify that all parsers allow empty configurations and return an empty list.

NTC-4957

### CODE REVIEW

File: tests/unit/test_parser.py [hunk 1]
```diff
@@ -13,17 +13,18 @@
...
-    for _file in glob.glob(f"{MOCK_DIR}/find_children_w_parents/{network_os}/*{TXT_FILE}"):
-        find_children_w_parents_parameters.append([_file, network_os])
+all_network_os = list(compliance.parser_map.keys())
+for _network_os in all_network_os:
+    for _file in glob.glob(f"{MOCK_DIR}/base/{_network_os}/*{TXT_FILE}"):
+        base_parameters.append([_file, _network_os])
+    for _file in glob.glob(f"{MOCK_DIR}/find_all_children/{_network_os}/*{TXT_FILE}"):
+        find_all_children_parameters.append([_file, _network_os])
+    for _file in glob.glob(f"{MOCK_DIR}/find_children_w_parents/{_network_os}/*{TXT_FILE}"):
+        find_children_w_parents_parameters.append([_file, _network_os])
```
Comment (joewesch) [inline tests/unit/test_parser.py]: This is just changing `network_os` to `_network_os` as the loop variable so that we don't perpetuate copy/pasting the same pylint disable comment over and over...

**Review by jeffkala — APPROVED** [review]
